### ISSUE TITLE
Support int8 transposed convolutions with per-channel weight quantization

### ISSUE THREAD

**Issue by lgeiger** [google/XNNPACK#2762]
TFLite uses int8 per-channel weight quantization for transposed convolutions.
While XNNPACK includes a fast transposed convolution operation it only supports per-tensor weight quantization (i.e. a single quantisation scale for the weight tensor), which means transposed convolutions in a TFLite QAT int8 model are currently not supported by XNNPACK and won't be accelerated.

It would be excellent if XNNPACK would add support for per-channel quantized weights to the transposed convolution op in the future to match the behaviour of the normal convolution.

**Comment by Maratyszcza** [comment]
Does TFLite support `TRANSPOSE_CONV` with per-channel quantization? Last time I looked at it, it wasn't supported there, thus I didn't implement it in XNNPACK.

**Comment by lgeiger** [comment]
> Does TFLite support `TRANSPOSE_CONV` with per-channel quantization?

Yes it does. `TRANSPOSE_CONV` kernels [supports both](https://github.com/tensorflow/tensorflow/blob/57cf4714fc1ee757b6f5226238035a981802b7d8/tensorflow/lite/kernels/transpose_conv.cc#L379-L380) but internally it will actually [convert it to per-channel](https://github.com/tensorflow/tensorflow/blob/57cf4714fc1ee757b6f5226238035a981802b7d8/tensorflow/lite/kernels/transpose_conv.cc#L384-L389) in both cases using [this code path](https://github.com/tensorflow/tensorflow/blob/57cf4714fc1ee757b6f5226238035a981802b7d8/tensorflow/lite/kernels/kernel_util.cc#L251-L257).

The MLIR converter will also [always output per-channel quantized weights](https://github.com/tensorflow/tensorflow/blob/57cf4714fc1ee757b6f5226238035a981802b7d8/tensorflow/compiler/mlir/lite/ir/tfl_ops.td#L713-L715) as far as I can tell.

**Comment by lgeiger** [comment]
> Does TFLite support TRANSPOSE_CONV with per-channel quantization?

@Maratyszcza I double check this again, and indeed TFLite supports per-channel quantization and the current converter will always generate per-channel quantized transposed convolutions when using QAT.
